minio/minio
# Pass multiple IDP config to console

## Description

This change passes sending configuration of all configured IDPs to console via a struct.

This change only changes the method of passing configuration to console and passes all enabled IDPs to console - further work is needed for integrating multiple IDPs in the web UI.

## Motivation and Context

This change allows us to integrate multiple IDPs in the web console in subsequent changes.

## How to test this PR?

Existing functionality should be preserved - i.e. a single IDP should continue to work as-is with this change.

Run `make podman-run` from https://github.com/minio/minio-iam-testing to start testing IDP services. 

Start `minio` with console on port 10000.

```shell
# Configure IDP as claim based provider (needs latest mc at time of writing)
$ mc admin idp set myminio openid     config_url="http://localhost:5557/dex/.well-known/openid-configuration"     client_id="minio-client-app-2"     client_secret="minio-client-app-secret-2"     scopes="openid,groups"     redirect_uri="http://127.0.0.1:10000/oauth_callback"     display_name="Login via dex2"     role_policy='' claim_name=groups

```

Now attempt to login normally via console (`dillon@example.io` and `dillon` as password).


## Types of changes

## Checklist:
